tests/test_proc/test_whitening.py

```diff
@@ -0,0 +1,141 @@
+        assert "time" in whitened_patch.dims
+        assert np.array_equal(
+            test_patch.coords.get_array("time"), whitened_patch.coords.get_array("time")
+        )
+        assert np.array_equal(
+            test_patch.coords.get_array("distance"),
+            whitened_patch.coords.get_array("distance"),
+        )
+
+    def test_default_whitening_2(self, test_patch):
+        """Ensure whitening can run without smoothing window size."""
+        whitened_patch = test_patch.whitening(freq_range=[5, 60])
+        assert "distance" in whitened_patch.dims
+        assert "time" in whitened_patch.dims
+        assert np.array_equal(
+            test_patch.coords.get_array("time"), whitened_patch.coords.get_array("time")
+        )
+        assert np.array_equal(
+            test_patch.coords.get_array("distance"),
+            whitened_patch.coords.get_array("distance"),
+        )
+
+    def test_default_whitening_3(self, test_patch):
+        """Ensure whitening can run without frequency range."""
+        whitened_patch = test_patch.whitening(freq_smooth_size=10)
+        assert "distance" in whitened_patch.dims
+        assert "time" in whitened_patch.dims
+        assert np.array_equal(
+            test_patch.coords.get_array("time"), whitened_patch.coords.get_array("time")
+        )
+        assert np.array_equal(
+            test_patch.coords.get_array("distance"),
+            whitened_patch.coords.get_array("distance"),
+        )
+
+    def test_edge_whitening(self, test_patch):
+        """Ensure whitening can run with edge cases frequency range."""
+        whitened_patch = test_patch.whitening(freq_range=[0, 50], freq_smooth_size=10)
+        assert "distance" in whitened_patch.dims
+        assert "time" in whitened_patch.dims
+        assert np.array_equal(
+            test_patch.coords.get_array("time"), whitened_patch.coords.get_array("time")
+        )
+        assert np.array_equal(
+            test_patch.coords.get_array("distance"),
+            whitened_patch.coords.get_array("distance"),
+        )
+
+        whitened_patch = test_patch.whitening(freq_range=[50, 100], freq_smooth_size=10)
+        assert "distance" in whitened_patch.dims
+        assert "time" in whitened_patch.dims
+        assert np.array_equal(
+            test_patch.coords.get_array("time"), whitened_patch.coords.get_array("time")
+        )
+        assert np.array_equal(
+            test_patch.coords.get_array("distance"),
+            whitened_patch.coords.get_array("distance"),
+        )
+
+    def test_single_freq_range_raises(self, test_patch):
+        """Ensure only one value for frequency range raises ParameterError."""
+        msg = "Frequency range must include two values"
+        freq_range = np.array([10])
+        with pytest.raises(ParameterError, match=msg):
+            test_patch.whitening(freq_range=freq_range, freq_smooth_size=3)
+
+    def test_freq_lt_0_raises(self, test_patch):
```

Comment:
Suggest changing the name to "test_freq_negative_raises"

Reply:
Done